### PR TITLE
libgostatic

### PR DESCRIPTION
**Pull request by zhuharev**
fix for #36 

Changes summary:

- all code moved to [lib](https://github.com/zhuharev/gostatic/tree/master/lib) path (but package named as `gostatic` - to discuss)
- `const VERSION` moved to [lib/version.go](https://github.com/zhuharev/gostatic/blob/master/lib/version.go)

**Processors**

- now `Processor` not struct, but interface{} ([view](https://github.com/zhuharev/gostatic/blob/master/lib/processors.go#L21-L25))
- processors moved to [processors](https://github.com/zhuharev/gostatic/tree/master/processors) path and created separated file to each processor

Also: less global variables and functions, this update **not breaking** for building and cli api

what's next
finnaly add testing and more [godoc](http://godoc.org/github.com/zhuharev/gostatic/lib)

### CODE REVIEW

**Comment by piranha** [comment]
I like what you did to processors! 

Although it seems `ignore` and `external` stopped working...

**Comment by piranha** [comment]
Ah, I see, they were not moved to the right place yet.

**Comment by piranha** [comment]
Ok, and it never detects that it doesn't need to re-render pages.

**Comment by zhuharev** [comment]
sorry, I'm inattentive

**Comment by zhuharev** [comment]
it should be tested

**Comment by zhuharev** [comment]
watcher algorithm is not changed, and work for me, can you describe the problem?

**Comment by piranha** [comment]
That's interesting, actually. I just get `Rendering 292 changed pages of 292 total` every time rather than having 0 of 292 on next run. This one is not about watcher.

File: lib/site.go [hunk 1]
```diff
@@ -89,11 +118,9 @@ func (site *Site) FindDeps() {
 func (site *Site) Process() int {
 	processed := 0
 	for _, page := range site.Pages {
-		if page.Changed() {
```
Comment (piranha) [inline lib/site.go]: I found the problem. :-)

File: lib/page.go [hunk 2]
```diff
@@ -132,16 +136,24 @@ func (page *Page) Is(path string) bool {
 	return page.Url() == path || page.Path == path
 }
 
+//will be used for dinamicly created pages
```
Comment (piranha) [inline lib/page.go]: "dynamically", and a space before word "will"

File: lib/page.go [hunk 3]
```diff
@@ -132,16 +136,24 @@ func (page *Page) Is(path string) bool {
...
+func (page *Page) WasRead(set ...bool) bool {
+	if len(set) > 0 {
```
Comment (piranha) [inline lib/page.go]: What's the point of this logic? Not sure I understand it.

File: lib/processors.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+		}
+		pre := ""
+		if p.Mode()&Pre != 0 {
+			pre = "(preprocessor)"
+		}
+		fmt.Printf("- %s %s\n\t%s\n", k, pre, p.Description())
+	}
+}
+
+func (s *Site) ProcessCommand(page *Page, cmd *Command, pre bool) error {
```
Comment (piranha) [inline lib/processors.go]: I guess those three functions should be removed?

File: lib/page.go (hunk 3 above)
Comment (zhuharev) [inline lib/page.go]: if `set` has value, page's unexported  field takes the value. Always return `page.wasread`. Changed in commit.

**Comment by piranha** [comment]
Cool! Now it builds my site without any difference to `master`, and stuff seems to be working. :)

**Pull request merged**